File: Rigged-Dice/rigged_dice.js
//
//   CSAcademy Problem - Rigged Dice
//   From: IEEEXtreme 15.0
//   Solved By: Dimitrios and Petros Papakonstantinou
//   Submission Score: 100/100 (1957ms - 36.6MB)
//

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let gameNum;
let currentGame = 0;
let n;
let currentRoll = 0;

function processGame() {
  let d1 = [], d2 = [];
  let aliceTotal = 0, bobTotal = 0;
  let currentAliceDie1 = true;

  function processRoll() {
    rl.question('', (line) => {
      const [alice, bob] = line.split(' ').map(Number);
      aliceTotal += alice;
      bobTotal += bob;

      if (currentAliceDie1) {
        d1.push(alice);
        d2.push(bob);
      } else {
        d2.push(alice);
        d1.push(bob);
      }

      if (aliceTotal !== bobTotal) {
        currentAliceDie1 = !currentAliceDie1;
      }

      currentRoll++;
      if (currentRoll < n) {
        processRoll();
      } else {
        const d1Count = d1.filter(x => x === 6).length;
        const d2Count = d2.filter(x => x === 6).length;

        console.log(d1Count > d2Count ? '1' : '2');

        currentGame++;
        if (currentGame < gameNum) {
          startNextGame();
        } else {
          rl.close();
        }
      }
    });
  }

  rl.question('', (input) => {
    n = parseInt(input);
    currentRoll = 0;
    processRoll();
  });
}

function startNextGame() {
  processGame();
}

rl.question('', (input) => {
  gameNum = parseInt(input);
  startNextGame();
});

File: Rigged-Dice/rigged_dice.py
#
#   CSAcademy Problem - Rigged Dice
#   From: IEEEXtreme 15.0
#   Solved By: Dimitrios and Petros Papakonstantinou
#   Submission Score: 100/100 (112ms - 776KB)
#

def process_game():
    n = int(input())
    d1, d2 = [], []
    alice_total, bob_total = 0, 0
    current_alice_die1 = True

    for _ in range(n):
        alice, bob = map(int, input().split())
        alice_total += alice
        bob_total += bob

        if current_alice_die1:
            d1.append(alice)
            d2.append(bob)
        else:
            d2.append(alice)
            d1.append(bob)

        if alice_total != bob_total:
            current_alice_die1 = not current_alice_die1

    d1_count = d1.count(6)
    d2_count = d2.count(6)

    print("1" if d1_count > d2_count else "2")

def main():
    game_num = int(input())
    for _ in range(game_num):
        process_game()

if __name__ == "__main__":
    main()
